### PR TITLE
Package some scripts of this repository in a Nix derivation

### PR DESCRIPTION
**Pull request by ptitfred**
Let's make some scripts packaged for Nix.

If you want to add it via an overlay, you can use this Nix expression:

```nix
self: super:

let fetchPackage = owner: repo: path: rev: sha256:
      self.callPackage (self.fetchFromGitHub { inherit owner repo rev sha256; } + path) {};
in {
     personal = {
        posix-toolbox = fetchPackage "ptitfred" "posix-toolbox" "/nix/default.nix"
          "2c4b1696e285daefe2bc7bf9336562bcb83b5da5" "10dps0rr0pv43xlkd1hiafhsallhji6ww3vca5cmq044m5zpx3di";
     };
   }
```

I personally have it under `$HOME/.config/nixpkgs/overlays/personal.nix` and this notably let me have those script in my path via home-manager:

```nix
{ pkgs, ... }:

{
  programs.git = {
    enable = true;
    userName = "Frédéric Menou";
    userEmail = "frederic.menou@gmail.com";
  };

  home.packages = with pkgs; [
    personal.posix-toolbox.git-bubbles
    personal.posix-toolbox.git-checkout-log
  ];
}
```

### CODE REVIEW

File: nix/release.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    postFixup = let
+      runtimePath = lib.makeBinPath buildInputs;
+    in
+    ''
+      sed -i "2 i export PATH=${runtimePath}:\$PATH" $out/bin/git-bubbles
+    '';
+
+    meta = {
+      homepage = https://github.com/ptitfred/posix-toolbox;
```
Comment (immae) [inline nix/release.nix:30]: strings without quotes are deprecated, you should wrap this one

**Comment by ptitfred** [comment]
As suggested in private by @immae, I should instead use a wrapper instead of fixuping the script.

File: nix/release.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, git, gnused }:
+
+let rev = "b5552cc32f567052dc8c5e6e3af84f8a2d254c35";
+    sha256 = "0gws4lajaylpx1vkq8kpq9q8sk0qil993n9y38xxfwvdmr5gj375";
+in
+  stdenv.mkDerivation rec {
+    name = "git-bubbles-${rev}";
+
+    src = fetchFromGitHub {
```
Comment (immae) [inline nix/release.nix:9]: Oh I didn’t see that, this could be replaced by `src = ./..`, this way you don’t have to specify a revision

**Comment by ptitfred** [comment]
For notes: I have this following overlay working:

```nix
self: super:

let fetchPackage = owner: repo: path: rev: sha256:
      self.callPackage (self.fetchFromGitHub { inherit owner repo rev sha256; } + path) {};
in {
     personal = {
        git-bubbles = fetchPackage "ptitfred" "posix-toolbox" "/nix/default.nix"
         "3f0cbc19ec36a0bc51031bc7d085bc826a159639" "1cs902sasc7d91gsl2xn0xyjkkka9hqwiz0icnjbssgvgjw8kxp3";
     };
   }
```

(pointing at this PR obviously for now)

**Comment by immae** [comment]
> For notes: I have this following overlay working:
> 
> ```nix
> self: super:
> 
> let fetchPackage = owner: repo: path: rev: sha256:
>       self.callPackage (self.fetchFromGitHub { inherit owner repo rev sha256; } + path) {};
> in {
>      personal = {
>         git-bubbles = fetchPackage "ptitfred" "posix-toolbox" "/nix/default.nix"
>          "3f0cbc19ec36a0bc51031bc7d085bc826a159639" "1cs902sasc7d91gsl2xn0xyjkkka9hqwiz0icnjbssgvgjw8kxp3";
>      };
>    }
> ```
> 
> (pointing at this PR obviously for now)
> 
> This is not very satisfying. I should rewrite this PR to produce a dict of the various scripts I would package at some point.

you probably want the release.nix rather than the default.nix. Or even better to follow usual conventions you should rename release.nix to default.nix and release.nix to default.nix

File: nix/release.nix (hunk 2 above)
Comment (ptitfred) [reply nix/release.nix:9]: Indeed, I wasn't sure about the best strategy as I'm not tagging any specific revision in this project for now.

Comment (ptitfred) [reply nix/release.nix]: Done, thanks for the tip.

**Comment by ptitfred** [comment]
> Or even better to follow usual conventions you should rename release.nix to default.nix and release.nix to default.nix

I've done that.

**Comment by ptitfred** [comment]
> As suggested in private by @immae, I should instead use a wrapper instead of fixuping the script.

This is done as well. I'm quite happy with the result ^^ !

**Comment by immae** [comment]
> > As suggested in private by @immae, I should instead use a wrapper instead of fixuping the script.
> 
> This is done as well. I'm quite happy with the result ^^ !

It looks good to me too :)

**Pull request merged**